### PR TITLE
Load car categories from db for user selection

### PR DESCRIPTION
**Pull request by tonia4ik-s**
In the modal window for adding a car, load the received categories for user selection, and accordingly create a car with the selected category.

Maryna made changes because the author of the PR could not do it alone for some reason

### CODE REVIEW

File: src/api/carCategory.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import instance from "./configurations";
+import {CAR_CATEGORY_URLS} from "../constants/url";
+
+export default class carCategoryService{
+    static getAllCategories()
+    {
+        return instance.get(CAR_CATEGORY_URLS.GET_CAR_CATEGORIES);
+    }
+}
```
Comment (marynakernychna) [inline src/api/carCategory.js]: Empty line at EoF

File: src/constants/url.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
-export const SERVER_URL = "https://localhost:44319/api";
+export const SERVER_URL = "https://localhost:5001/api";
```
Comment (marynakernychna) [inline src/constants/url.js]: We have another port for service

File: src/constants/url.js [hunk 3]
```diff
@@ -18,3 +19,7 @@ export const CAR_URLS = {
     ADD_CAR: CAR_URL + "/add",
     GET_USER_CARS: CAR_URL + "/user-cars"
 };
+
+export const CAR_CATEGORY_URLS = {
+    GET_CAR_CATEGORIES: CAR_CATEGORY_URL + "/categories"
```
Comment (marynakernychna) [inline src/constants/url.js]: The path does not work

**Review by marynakernychna — CHANGES_REQUESTED** [review]
> The path to the service is incorrect, so I can't finish the review
> Categories, respectively, are not written in the drop-down list, through the first point
> Please also return the deleted file package-lock.json

File: src/components/newCarModal/addCar.js [hunk 4]
```diff
@@ -41,15 +59,28 @@ class NewCarModal extends React.Component {
...
                 >
                     <Form.Item name="model"
+                               label={"Model: "}
                                rules={[
                                    {
                                        type: "string",
-                                       pattern: new RegExp("^[a-zA-Z\\d ]*$"),
+                                       pattern: new RegExp("^[A-Z]"),
+                                       message:
+                                       carErrorMessages.CANNOT_START_WITH_SPACE_MESSAGE
```
Comment (marynakernychna) [inline src/components/newCarModal/addCar.js]: Uncorrent error message

Screenshot 2022-05-17 153213

File: src/components/newCarModal/addCar.js [hunk 5]
```diff
@@ -60,9 +91,10 @@ class NewCarModal extends React.Component {
                                    },
                                ]}
                     >
-                        <Input addonBefore="Model:" placeholder="Model"/>
+                        <Input /*addonBefore="Model:"*/ placeholder="Model"/>
```
Comment (marynakernychna) [inline src/components/newCarModal/addCar.js]: I think you can delete the comment

File: src/services/carCategoryService.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+export async function getCarCategories() {
+    return carCategoryService
+        .getAllCategories()
+        .then(
+            async (response) => {
+                return await response.data["carCategories"];
+            },
+            (err) => {
+                store.dispatch(closeModal());
+                err.response.status === 404
```
Comment (marynakernychna) [inline src/services/carCategoryService.js:17]: I think you should show a message for any error

File: src/services/carService.js [hunk 7]
```diff
@@ -20,16 +20,21 @@ export function addCar(values) {
             () => {
                 successMessage(carErrorMessages.CAR_ADDED_SUCCESSFUL);
             },
-            () => {
-                errorMessage(
-                    carErrorMessages.CAR_ADDING_FAILED,
-                    generalErrorMessages.SOMETHING_WENT_WRONG
-                );
+            (err) => {
+                err.response.status === 406
```
Comment (marynakernychna) [inline src/services/carService.js:24]: I think you should display a message for any error

**Review by marynakernychna — DISMISSED** [review]

**Review by marynakernychna — APPROVED** [review]

**Review by Grasmus — APPROVED** [review]

**Pull request merged**